src/EVMBlackjack.sol

```diff
@@ -0,0 +1,337 @@
+        if (game.state != State.WAITING_FOR_RANDOMNESS) {
+            // We can't fulfill randomness if not waiting for randomness.
+            revert InvalidStateTransition();
+        } else if (game.shoeCount == SHOE_STARTING_COUNT) {
+            // seed
+            // pair of aces -- player gets AA (13, 26)
+            // pair         -- player gets 77 (6, 19)
+            // other        -- player gets 54 (4, 16)
+
+            // ace          -- dealer gets A  (0)
+            // other        -- dealer gets 2  (1)
+
+            // We are at initial deal state, so
+            // Deal player card 1,
+            if (_randomness == keccak256("pair of aces")) {
+                // TEMP
+                dealPlayerCard(_player, 13, 0);
+            } else if (_randomness == keccak256("pair")) {
+                dealPlayerCard(_player, 6, 0);
+            } else {
+                dealPlayerCard(_player, 4, 0);
+            }
+
+            // Deal dealer card 1,
+            if (_randomness == keccak256("ace")) {
+                // TEMP
+                dealDealerCard(_player, 0);
+                game.state = State.READY_FOR_INSURANCE;
+            } else {
+                dealDealerCard(_player, 1);
+                game.state = State.READY_FOR_PLAYER_ACTION;
+            }
+
+            // Deal player card 2,
+            if (_randomness == keccak256("pair of aces")) {
+                // TEMP
+                dealPlayerCard(_player, 26, 0);
+            } else if (_randomness == keccak256("pair")) {
+                dealPlayerCard(_player, 19, 0);
+            } else {
+                dealPlayerCard(_player, 16, 0);
+            }
+
+            // Update shoe.
+            game.shoeCount -= 3;
+        } else if (game.lastAction == Action.SPLIT_ACES) {
+            // Player's last action was split aces, so we handle and go to Dealer Action.
+            dealPlayerCard(_player, 50, 0);
+            dealPlayerCard(_player, 51, 0);
+
+            game.shoeCount -= 2;
+            game.state = State.DEALER_ACTION;
+        } else if (game.lastAction == Action.SPLIT) {
+            // Player's last action was split, so we handle and go to Ready for Player Action.
+            dealPlayerCard(_player, 50, 0);
+            dealPlayerCard(_player, 51, 0);
+
+            game.shoeCount -= 2;
+            game.state = State.READY_FOR_PLAYER_ACTION;
+        } else if (game.lastAction == Action.DOUBLE_DOWN) {
+            // Player's last action was double down, so we handle and go to Dealer Action.
+            dealPlayerCard(_player, 15, 0);
+
+            game.shoeCount--;
+            game.state = State.DEALER_ACTION;
+        } else if (game.lastAction == Action.HIT) {
+            // Player's last action was hit, so we handle and go to Ready for Player Action.
+            dealPlayerCard(_player, 25, 0);
+
+            game.shoeCount--;
+            game.state = State.READY_FOR_PLAYER_ACTION;
+        } else {
+            //
```

Comment:
stand?

Reply:
May be a bad state?

Reply:
up to have bad state/default fall-through. but fb was more that we need an else if for STAND action

Reply:
Sounds good, will do